client/src/components/export/dialogs/spoolExportDialog.tsx

```diff
@@ -0,0 +1,510 @@
+import { Col, Collapse, Form, Input, InputNumber, Modal, Radio, Switch } from "antd";
+import { ISpool } from "../../../pages/spools/model";
+import { useSavedState } from "../../../utils/saveload";
+import { CSVExportOptions, QRExportOptions, exportAsCSV, exportQRCode } from "../functions";
+import { useTranslate } from "@refinedev/core";
+import { RadioChangeEvent } from "antd/lib";
+import { numberFormatter, numberParser } from "../../../utils/parsing";
+
+interface SpoolExportDialog {
+  visible: boolean;
+  items: ISpool[];
+  onCancel: () => void;
+}
+
+const SpoolExportDialog: React.FC<SpoolExportDialog> = ({ visible, items, onCancel }) => {
+  const t = useTranslate();
+
+  const [qrForm] = Form.useForm();
+  const [csvForm] = Form.useForm();
+
+  const [exportType, setExportType] = useSavedState<"QR-Code" | "CSV">("export-exportType", "QR-Code");
+  const [csvDefaultCollapse, setCSVDefaultCollapse] = useSavedState<string[]>("export-collapseState-csv-spool", []);
+
+  const [csvExportOptions, setCSVExportOptions] = useSavedState<CSVExportOptions<ISpool>>("export-CSVOptions-spool", {
+    delimiter: ";",
+    includeHeaders: true,
+    filename: "Spool Export",
+    id: true,
+    location: true,
+    remaining_length: true,
+    used_length: true,
+  });
+  const [qrExportOptions, setQRExportOptions] = useSavedState<QRExportOptions>("export-qrOptions-spool", {
+    boxSize: 32,
+    padding: 2,
+    useFullURL: false,
+  });
+
+  const transformSpoolToQRContent = (spool: ISpool) => {
+
+    let content = `web+spoolman:s-${spool.id}`;
```

Comment:
The QR code generation and parsing should be done in a common shared location,\.